lib/aws/client.go

```diff
@@ -0,0 +1,46 @@
+package aws
+
+import (
+	"sync"
+
+	"github.com/GetTerminus/s3helper/lib/parser"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+type client struct {
+	sess *session.Session
+	once sync.Once
+}
+
+// Session returns an authenticated AWS session, used to make API calls.
+func (c *client) GetSession() *session.Session {
+
+	// Only ever create one session
+	c.once.Do(func() {
+		// Create a session, session.Must should handle any errors
+		// https://docs.aws.amazon.com/sdk-for-go/api/aws/session/#Must
+		c.sess = session.Must(
```

Comment:
This is likely a violation of "Don't Panic". Generally speaking, you only want to panic in your `main.go` directly. All other cases would return an error of some sort. (I suspect this `GetSession` is only called in your `main.go` but it _could_ be called elsewhere as structured.)